src/main/java/edu/kis/powp/jobs2d/command/CompoundCommand.java

```diff
@@ -0,0 +1,42 @@
+package edu.kis.powp.jobs2d.command;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class CompoundCommand implements ICompoundCommand {
+
+	private String name;
+	private List<DriverCommand> driverCommands;
+
+	public CompoundCommand(List<DriverCommand> driverCommands, String name) {
+		this.driverCommands = driverCommands;
+		this.name = name;
+	}
+
+	@Override
+	public void execute(Job2dDriver driver) {
+		driverCommands.forEach((c) -> c.execute(driver));
+	}
+
+	@Override
+	public Iterator<DriverCommand> iterator() {
+		return driverCommands.iterator();
+	}
+
+	@Override
+	public CompoundCommand clone() {
+		return new CompoundCommand(new ArrayList<>(driverCommands), name);
```

Comment:
It was supposed to be a "deep copy". I see that you reuse the same driversCommand list.